Realm/RealmSwift/List.swift

```diff
@@ -105,50 +197,115 @@ public final class List<T: Object>: ListBase, SequenceType {
         }
     }
 
-    // MARK: Mutating
+    // MARK: Mutation
+
+    /**
+    Appends the given object to the end of the list.
 
+    :warning: This method can only be called during a write transaction.
+
+    :param: object An object.
+    */
     public func append(object: T) {
         _rlmArray.addObject(unsafeBitCast(object, RLMObject.self))
     }
 
-    public func append<S where S: SequenceType>(objects: S) {
-        for obj in objects {
-            _rlmArray.addObject(unsafeBitCast(obj as T, RLMObject.self))
+    /**
+    Appends the objects in the given sequence to the end of the list.
+
+    :warning: This method can only be called during a write transaction.
+
+    :param: objects A sequence of objects.
+    */
+    public func append<S: SequenceType where S.Generator.Element == T>(objects: S) {
```

Comment:
All the relevant details are in the type constraint, but I can break it down here if you like:

`S: SequenceType where S.Generator.Element == T` means that `objects` can only be a type that conforms to `SequenceType` where its elements are all the same type as the objects already stored in this `List`. So if someone creates a custom `LinkedList<Object>` type that conforms to `SequenceType`, this will just work.